CAIPs/caip-372.md

```diff
@@ -0,0 +1,124 @@
+---
+caip: 372
+title: Wallet Information Metadata Standard
+author: Pedro Gomes (@pedrouid)
+discussions-to: https://github.com/ChainAgnostic/CAIPs/pull/372
+status: Draft
+type: Standard
+created: 2025-08-11
+---
+
+## Simple Summary
+
+This CAIP defines a standard `WalletInfo` metadata object to identify and validate the wallet provider that a client application is interacting with. It can be used across RPC methods, APIs, and protocols to provide consistent, cross-chain wallet identification.
+
+## Abstract
+
+The `WalletInfo` object provides a minimal, standardized set of fields that uniquely identify a wallet provider and allow dApps to confirm they are connected to the intended wallet. The object is designed for use in CAIP-25 session properties, CAIP-282 messages, and other API contexts. The definition includes a UUIDv4 identifier, human-readable name, icon URI, and reverse-DNS string to support clear and reliable wallet identification.
+
+## Motivation
+
+Wallet providers often expose their brand, capabilities, or identity inconsistently across blockchains and APIs. This creates friction for dApps attempting to present clear wallet branding or confirm that the connection is genuine.
+
+By standardizing `WalletInfo`, wallet providers can:
+
+- Ensure a consistent presentation across dApps and chains.
+- Reduce risk of misidentification or spoofing by unverified wallets.
+- Facilitate richer integrations in protocols like CAIP-25 and CAIP-282.
+
+For dApps, this enables:
+
+- Trust signals to the user (correct wallet branding and identity).
+- Improved debugging and support workflows.
+- Safer handling of wallet interactions across chains.
+
+## Specification
+
+A `WalletInfo` object MUST contain the following fields:
+
+- **uuid**: A globally unique identifier (UUIDv4) generated by the wallet provider to uniquely distinguish different wallet provider sessions that have otherwise matching properties. The cryptographic uniqueness of UUIDv4 ensures separate identification of independent sessions.
```

Comment:
if these are identifying sessions, wouldn't a time-sortable UUID like v6 or v7 be justified here? or is there some danger in leaking temporal/sequence information in combination with the other properties?